wear/src/main/java/io/homeassistant/companion/android/phone/PhoneSettingsListener.kt

```diff
@@ -0,0 +1,96 @@
+package io.homeassistant.companion.android.phone
+
+import android.net.Uri
+import android.util.Log
+import com.google.android.gms.tasks.Tasks
+import com.google.android.gms.wearable.DataClient
+import com.google.android.gms.wearable.DataEvent
+import com.google.android.gms.wearable.DataEventBuffer
+import com.google.android.gms.wearable.DataMap
+import com.google.android.gms.wearable.DataMapItem
+import com.google.android.gms.wearable.MessageEvent
+import com.google.android.gms.wearable.PutDataMapRequest
+import com.google.android.gms.wearable.Wearable
+import com.google.android.gms.wearable.WearableListenerService
+import dagger.hilt.android.AndroidEntryPoint
+import io.homeassistant.companion.android.common.data.integration.IntegrationRepository
+import kotlinx.coroutines.runBlocking
+import javax.inject.Inject
+
+@AndroidEntryPoint
+class PhoneSettingsListener : WearableListenerService(), DataClient.OnDataChangedListener {
+
+    @Inject
+    lateinit var integrationUseCase: IntegrationRepository
+
+    companion object {
+        private const val TAG = "PhoneSettingsListener"
+    }
+
+    override fun onCreate() {
+        super.onCreate()
+        Log.d(TAG, "We have favorite message listener")
+    }
+
+    override fun onMessageReceived(event: MessageEvent) {
+        Log.d(TAG, "Message received: $event")
+        if (event.path == "/send_home_favorites") {
+            val nodeId = event.sourceNodeId
+            sendHomeFavorites(nodeId)
+        } else if (event.path == "/save_home_favorites")

```

Comment:
Do we need the "/save_home_favorites" message? If the data is put in the DataClient, we will be notified in the onDataChanged listener, right? Or is there a scenario where we do get this message but not the data changed notification?

Reply:
Honestly I thought we needed both but I very well could be mistaken lol

Reply:
yup you are right we are not using `sendMessage` with `/save_home_favorites`  so this should not be needed, I was just trying to very closely follow what was laid out before :)

Reply:
Yeah with the onboarding I was struggling a bit as well. In the end I think I might have overdone it a bit there as well. I wanted to get anything already in the DataClient, but then also send a message to the phone to send it again. The idea was that it would show what is available ASAP, but still update with the latest info. I could have a look to see if the onboarding one can be simplified as well

Reply:
yup this line was not needed, but we still need the reference in the manifest